tools/PsesLogAnalyzer/Parse-PsesLog.ps1

```diff
@@ -0,0 +1,268 @@
+                    ($line -match '^\s+Writing Response ''(?<msg>[^'']+)'' with id (?<id>\d+)')) {
+                $result.LogMessageType = [PsesLogMessageType]::Response
+                $msg = $matches["msg"]
+                $id = $matches["id"]
+                $json = parseLogMessageBodyAsJson
+                $result.LogMessage = [PsesJsonRpcMessage]::new($msg, $id, $json.Data, $json.DataSize)
+            }
+            elseif (($Method -eq 'WriteMessage') -and
+                    ($line -match '^\s+Writing event ''(?<msg>[^'']+)''')) {
+                $result.LogMessageType = [PsesLogMessageType]::Notification
+                $msg = $matches["msg"]
+                $json = parseLogMessageBodyAsJson
+                $result.LogMessage = [PsesNotificationMessage]::new($msg, [PsesNotificationSource]::Server, $json.Data, $json.DataSize)
+            }
+            else {
+                if  ($line -match '^\s+Exception: ') {
+                    $result.LogMessageType = [PsesLogMessageType]::Exception
+                }
+                elseif  ($line -match '^\s+Handled exception: ') {
+                    $result.LogMessageType = [PsesLogMessageType]::HandledException
+                }
+                else {
+                    $result.LogMessageType = [PsesLogMessageType]::Log
+                }
+
+                $body = parseLogMessageBody $line
+                $result.LogMessage = [PsesLogMessage]::new($body)
+            }
+
+            $result
+        }
+
+        function parseLogMessageBody([string]$startLine = '', [switch]$Discard) {
+            if (!$Discard) {
+                $strBld = [System.Text.StringBuilder]::new($startLine, 4096)
+                $newLine = "`r`n"
+            }
+
+            try {
+                while ($true) {
+                    $peekLine = peekLine
+                    if ($null -eq $peekLine) {
+                        break
+                    }
+
+                    if (($peekLine.Length -gt 0) -and ($peekLine[0] -ne ' ') -and ($peekLine -match $logEntryRegex)) {
+                        break
+                    }
+
+                    $nextLine = nextLine
+                    if (!$Discard) {
+                        [void]$strBld.Append($nextLine).Append($newLine)
+                    }
+                }
+            }
+            catch {
+                Write-Error "Failed parsing message body with error: $_"
+            }
+
+            if (!$Discard) {
+                $msgBody = $strBld.ToString().Trim()
+                $msgBody
+            }
+            else {
+                $startLine
+            }
+        }
+
+        function parseLogMessageBodyAsJson() {
+            $result = [PSCustomObject]@{
+                Data = $null
+                DataSize = 0
+            }
+
+            $obj = $null
+
+            if ($SkipRpcMessageBody) {
+                parseLogMessageBody -Discard
+                return $result
+            }
+            else {
```

Comment:
`else` here is not needed, since `if` always returns

Reply:
Fixed